build/build2.mk

```diff
@@ -14,12 +14,13 @@
 #  | Author: Sascha Schumann <sascha@schumann.cx>                         |
 #  +----------------------------------------------------------------------+
 
-include generated_lists
-
 config_h_in = main/php_config.h.in
 
 targets = configure $(config_h_in)
 
+m4_files.exec = echo TSRM/*.m4 Zend/*.m4 ext/*/config*.m4 sapi/*/config*.m4
+m4_files = $(shell $(m4_files.exec))$(m4_files.exec:sh)
```

Comment:
Do you have any reference on what this means?

Reply:
`m4_files.exec` can be named `anything` and it's value is just a string of a command to be executed.

`m4_files` is that command executed and its output captured.

Main issue here is portability. On GNU Make the `$(shell executable-command)` works, on every other Make derivative (for example, Solaris or some BSD speciality) the `$(executable-command:sh)` works. However, both are smart enough to produce an empty result on each other's implementations. So, a combination of both works on probably most if not all systems. However, this is still not POSIX. POSIX make implementation is so limited actually that capturing command output cannot be done like this. Therefore this *hack* and simplification.

Something like this in short.

There is then another fix/improvement coming up to ditch the `?=` operators into a simple `=` and that $(SUPRESS_WARNINGS) updated for current autoconf outputs... But this is another commit, not related to this directly.

Reply:
Ok, taking POSIX stuff a bit back. POSIX compatible implementation might just be this:

M4_FILES variable generated and passed to make in the buildconf script directly... Even simpler and less hacks like this.